### PR TITLE
Document how to use react-hot-loader in production

### LINKED ISSUE #1104: Document that react-hot-loader/babel should now be used in production too
In #1081 the RHL Babel plugin for when `process.env.NODE_ENV` is `production` was changed from a no-op to a plugin that removes the `hot()` wrapper.

This means it's now recommended to not only use the plugin in development, but in any production configs too.

As such, it would be good to update the docs to say that `react-hot-loader/babel` should be used in all environments. eg here:
https://github.com/gaearon/react-hot-loader#getting-started
https://github.com/gaearon/react-hot-loader#migrating-from-create-react-app

CC @smelukov

### PR DESCRIPTION
**Pull request by silvenon**
Fixes #1104.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,14 +17,15 @@ Watch
 **[Dan Abramov's talk on Hot Reloading with Time Travel](https://www.youtube.com/watch?v=xsSnOQynTHs).**
 
 # Deprecation note
+
 React-Hot-Loader was your friendly neightbour, living outside of React. But it was limiting it's powers and causing not the greatest experience. It's time to make a next step.
 
-__React-Hot-Loader is expected to be replaced by [React Fast Refresh](https://github.com/facebook/react/issues/16604)__. Please remove React-Hot-Loader is Fast Refresh is supported at your environment.
+**React-Hot-Loader is expected to be replaced by [React Fast Refresh](https://github.com/facebook/react/issues/16604)**. Please remove React-Hot-Loader is Fast Refresh is supported at your environment.
```
Comment (theKashey) [inline README.md]: :coneofshame: React-Hot-Loader __if__ Fast Refresh is supported at your environment

File: README.md [hunk 2]
```diff
@@ -161,6 +162,20 @@ export default hot(module)(App);
...
+
+The webpack patch, `hot`, Babel plugin, `@hot-loader/react-dom` etc. are all safe to use in production; they leave a minimal footprint, so there is no need to complicate your configuration based on the environment. Using the Babel plugin in production is even recommended because it switches to cleanup mode.
+
+Just ensure that the production mode has been properly set, both as an environment variable and in your bundler. E.g. with webpack you would build your code by running something like:
+
+```
+NODE_ENV=production webpack --mode production
+```
+
+`NODE_ENV=production` is needed for the Babel plugin, while `--mode production` uses [`webpack.DefinePlugin`](https://webpack.js.org/plugins/define-plugin/) to set `process.env.NODE_ENV` inside the compiled code itself, which is used by `hot` and `@hot-loader/react-dom`.
```
Comment (theKashey) [inline README.md]: I am not sure `--mode production` does it. At least there is nothing about this in docs.

File: README.md (hunk 1 above)
Comment (silvenon) [reply README.md]: ✅

File: README.md (hunk 2 above)
Comment (silvenon) [reply README.md]: It does, I tested it. Try it for yourself.

```js
// webpack.config.js
module.exports = {
  entry: './src',
}
```
```js
// src/index.js
console.log(process.env.NODE_ENV)
```
```sh
yarn add webpack webpack-cli
yarn webapck --mode production
```
```js
// dist/index.js
// ...among bunch of stuff...
console.log('production')
```

**Comment by theKashey** [comment]
👍

**Pull request merged**